test/tests/parallel/set3/container_configuration/env_vars_from_files.bats

```diff
@@ -0,0 +1,79 @@
+load "${REPOSITORY_ROOT}/test/helper/setup"
+load "${REPOSITORY_ROOT}/test/helper/common"
+
+# Feature (ENV value sourced from file):
+# - An ENV with a `__FILE` suffix will read a value from a referenced file path to set the actual ENV (assuming it is empty)
+# - Feature implemented at: `variables-stack.sh:__environment_variables_from_files()`
+# - Feature PR: https://github.com/docker-mailserver/docker-mailserver/pull/4359
+
+BATS_TEST_NAME_PREFIX='[Configuration] (ENV __FILE support) '
+CONTAINER1_NAME='dms-test_env-files_success'
+CONTAINER2_NAME='dms-test_env-files_warning'
+CONTAINER3_NAME='dms-test_env-files_error'
+
+function setup_file() {
+  export CONTAINER_NAME
+  export FILEPATH_VALID='/tmp/file-with-value'
+  export FILEPATH_INVALID='/path/to/non-existent-file'
+
+  export FILE_WITH_VALUE="${TEST_TMP_CONFIG}/test_secret"
+  echo 1 > "${FILE_WITH_VALUE}"
+
+  # ENV is set via file content (valid file path):
+  CONTAINER_NAME=${CONTAINER1_NAME}
+  _init_with_defaults
+  local CUSTOM_SETUP_ARGUMENTS=(
+    --env ENABLE_POP3__FILE="${FILEPATH_VALID}"
+    -v "${FILE_WITH_VALUE}:${FILEPATH_VALID}"
+  )
+  _common_container_setup 'CUSTOM_SETUP_ARGUMENTS'
+
+  # ENV is already set explicitly, logs warning:
+  CONTAINER_NAME=${CONTAINER2_NAME}
+  _init_with_defaults
+  local CUSTOM_SETUP_ARGUMENTS=(
+    --env TEST_ENV="manual-secret"
+    --env TEST_ENV__FILE="${FILEPATH_VALID}"
+  )
+  _common_container_setup 'CUSTOM_SETUP_ARGUMENTS'
```

Comment:
Should the file still be mounted? I understand it shouldn't be relevant for the current logic, but for the path to be valid we could better match the first test case?

```suggestion
  # ENV is already set explicitly, a warning should be logged:
  CONTAINER_NAME=${CONTAINER2_NAME}
  _init_with_defaults
  local CUSTOM_SETUP_ARGUMENTS=(
    --env ENABLE_POP3="0"
    --env ENABLE_POP3__FILE="${FILEPATH_VALID}"
    -v "${FILE_WITH_VALUE}:${FILEPATH_VALID}"
  )
  _common_container_setup 'CUSTOM_SETUP_ARGUMENTS'
```